GVFS/GVFS.Service/Handlers/CommonMessagesHandler.cs

```diff
@@ -0,0 +1,78 @@
+﻿using GVFS.Common.NamedPipes;
+using GVFS.Common.Tracing;
+using System.Runtime.Serialization;
+
+namespace GVFS.Service.Handlers
+{
+    public static class CommonMessagesHandler
+    {
+        public static bool HandleMessage(
+            ITracer tracer,
+            string etwArea,
+            RepoRegistry repoRegistry,
+            NamedPipeMessages.Message message,
+            NamedPipeServer.Connection connection)
+        {
+            bool isHandled = false;
+
+            switch (message.Header)
+            {
+                case NamedPipeMessages.RegisterRepoRequest.Header:
+                    try
+                    {
+                        NamedPipeMessages.RegisterRepoRequest mountRequest = NamedPipeMessages.RegisterRepoRequest.FromMessage(message);
+                        RegisterRepoHandler mountHandler = new RegisterRepoHandler(tracer, repoRegistry, connection, mountRequest);
+                        mountHandler.Run();
+                        isHandled = true;
+                    }
+                    catch (SerializationException ex)
+                    {
+                        tracer.RelatedError("Could not deserialize mount request: {0}", ex.Message);
+                    }

```

Comment:
Can we find a way to have only one `try { } catch { }` around everything by setting a `string errorMessage = ` at the beginning of each `case` statement and use that in the catch or maybe if the user isn't going to see this message and is only in the logs we could find a way to use the type of the `MessageHandler`?  That would take more refactoring though.

Reply:
replaced multiple `try...catch` blocks with single higher level one.